Allow data to be either "long" or "tall"

peterfranciscook opened the pull request:
1. Allow datetime input class for x data.
2. Allow data to be either "long" or "tall." "Long" data is defined as a large number of increasing x-points and is reduced based on plot width. "Tall" data is defined as a large number of increasing y-points and is reduced based on plot height.

peterfranciscook committed:
Allow data to be either "long" or "tall"

1. Allow datetime input class for x data.
2. Allow data to be either "long" or "tall." "Long" data is defined as a large number of increasing x-points and is reduced based on plot width. "Tall" data is defined as a large number of increasing y-points and is reduced based on plot height.

tuckermcclure commented:
Thanks @peterfranciscook. I see that this works with the built in plot zoom tools. There are some things holding me back from merging it, however. If you wanted to add these things, then great! Otherwise, next time I work on this tool, I can add them.

* There are no examples of the updated "tall" data in the `example_of_line_plot_reducer.m` file.
* The included `LinePlotExplorer` is still only for the x axis; I think that would need to be updated for consistency, and this would go well with adding the examples above.
* I'm seeing some weirdness using MATLAB's built-in plot exploration tools to zoom in and out.
* The method of detecting which is the "long" or "tall" axis can't tell which is which when they each increase monotonically. I think we'd need some kind of input argument (or even just a different plot function!) to specify which axis contains the independent variable.

Thanks for the PR!

peterfranciscook committed:
Reduce the Number of Figure Mouse Callbacks & Listener Calls

Events firing too often were having a negative effect on timeSeriesTimeRange objects
